### PR TITLE
[runtime env] [Doc] add more details about runtime env logs

### PR DESCRIPTION
**Pull request by architkulkarni**
## Why are these changes needed?
Clarifies the logging behavior for runtime envs, and adds the runtime env logs fileto the list of log files in the main logging page.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-core/handling-dependencies.rst [hunk 1]
```diff
@@ -601,8 +601,8 @@ the error message in detail.
     a = A.options(runtime_env=bad_env).remote()
     ray.get(a.f.remote())
 
-You can also enable runtime_env debugging logs by setting an environment variable ``RAY_RUNTIME_ENV_LOG_TO_DRIVER_ENABLED=1``. 
-It will print the full runtime_env setup log messages to the driver.
-The same information can be found from the log file ``runtime_env*.log`` from the log directory. 
+You can also enable ``runtime_env`` debugging log streaming by setting an environment variable ``RAY_RUNTIME_ENV_LOG_TO_DRIVER_ENABLED=1`` before starting Ray on the head node (e.g. with ``ray start --head``).
```
Comment (architkulkarni) [inline doc/source/ray-core/handling-dependencies.rst:604]: @rkooo567 let me know if this is correct -- I wasn't 100% sure if it needs to be set on all the nodes, or just the head node.

Comment (rkooo567) [reply doc/source/ray-core/handling-dependencies.rst:604]: It should be on all nodes. In a worker, it should also be set

File: doc/source/ray-core/handling-dependencies.rst [hunk 2]
```diff
@@ -601,8 +601,8 @@ the error message in detail.
     a = A.options(runtime_env=bad_env).remote()
     ray.get(a.f.remote())
 
-You can also enable runtime_env debugging logs by setting an environment variable ``RAY_RUNTIME_ENV_LOG_TO_DRIVER_ENABLED=1``. 
-It will print the full runtime_env setup log messages to the driver.
-The same information can be found from the log file ``runtime_env*.log`` from the log directory. 
+You can also enable ``runtime_env`` debugging log streaming by setting an environment variable ``RAY_RUNTIME_ENV_LOG_TO_DRIVER_ENABLED=1`` before starting Ray on the head node (e.g. with ``ray start --head``). 
+This will print the full ``runtime_env`` setup log messages to the driver (the script that calls ``ray.init()``).
```
Comment (rkooo567) [inline doc/source/ray-core/handling-dependencies.rst:605]: Can you add an example underneath?

File: doc/source/ray-core/handling-dependencies.rst [hunk 3]
```diff
@@ -601,8 +601,8 @@ the error message in detail.
...
 
-Look :ref:`Logging Directory Structure <logging-directory-structure>` for more details.
+Regardless of the value of this environment variable, the same information can be found in the log file ``runtime_env_setup-[job_id].log`` from the log directory.
```
Comment (rkooo567) [inline doc/source/ray-core/handling-dependencies.rst:607]: I think there are other cases too. Can you specify all? 

- job id
- client
- per actor / task

File: doc/source/ray-observability/ray-logging.rst [hunk 4]
```diff
@@ -127,6 +127,7 @@ Here's a Ray log directory structure. Note that ``.out`` is logs from stdout/std
 - ``redis.[out|err]``: A log file of GCS (Redis by default).
 - ``worker-[worker_id]-[job_id]-[pid].[out|err]``: Python/Java part of Ray drivers and workers. All of stdout and stderr from tasks/actors are streamed here. Note that job_id is an id of the driver.
 - ``io-worker-[worker_id]-[pid].[out|err]``: Ray creates IO workers to spill/restore objects to external storage by default from Ray 1.3+. This is a log file of IO workers.
+- ``runtime_env_setup-[job_id].log``: Logs from installing :ref:`runtime environments<runtime-environments>`.  This file will only be present if a runtime environment is installed.
```
Comment (rkooo567) [inline doc/source/ray-observability/ray-logging.rst:130]: same com,ent as above

**Comment by architkulkarni** [comment]
Great feedback, I addressed the comments.  This is ready for another review @rkooo567, sorry for the delay!

**Review by rkooo567 — APPROVED** [review]

**Comment by rkooo567** [comment]
No problem! Thanks @architkulkarni !

**Comment by rkooo567** [comment]
Lmk when it is ready to merge

**Comment by architkulkarni** [comment]
@rkooo567 Ready to merge!

**Pull request merged**